src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1531,7 +1531,30 @@ function ($a) {
                         ->arrayNode('stop_worker_on_signals')
                             ->defaultValue([])
                             ->info('A list of signals that should stop the worker; defaults to SIGTERM and SIGINT.')
-                            ->integerPrototype()->end()
+                            ->beforeNormalization()
+                                ->always()
+                                ->then(function ($v) {
+                                    return array_map(static function ($element): int {
+                                        $element = \is_array($element) ? $element['signal'] : $element;
+                                        if (\is_string($element)) {
+                                            try {
+                                                $signal = \constant($element);
+                                            } catch (\Error) {
+                                                $signal = null;
+                                            }
+
+                                            if (null === $signal || !str_starts_with($element, 'SIG') || !\array_key_exists($element, get_defined_constants(true)['pcntl'])) {
```

Comment:
what about NOT doing any validation but just opportunistically turn SIG* into their value when defined?
the benefit of not validating is that this would allow env vars to be used here

Reply:
That's updated! I removed the validation and updated PR description. :slightly_smiling_face: